minigalaxy/download_manager.py

```diff
@@ -5,19 +5,43 @@
 from minigalaxy.config import Config
 from minigalaxy.constants import DOWNLOAD_CHUNK_SIZE, MINIMUM_RESUME_SIZE, SESSION
 from minigalaxy.download import Download
+from minigalaxy.paths import CACHE_DIR
 
 
 class __DownloadManger:
+    @staticmethod
+    def get_availablediskspace(location):
+        """Check disk space available to the user. This method uses the absolute path so
+        symlinks to disks with sufficient space are correctly measured. Note this is
+        a linux-specific command."""
+        absolute_location = os.path.realpath(location)
+        disk_status = os.statvfs(os.path.dirname(absolute_location))
+        available_diskspace = disk_status.f_frsize * disk_status.f_bavail
+        return available_diskspace
+
     def __init__(self):
         self.__queue = queue.Queue()
         self.__current_download = None
         self.__cancel = False
         self.__paused = False
+        self.window = None
 
         download_thread = threading.Thread(target=self.__download_thread)
         download_thread.daemon = True
         download_thread.start()
 
+    def check_diskspace(self, file_size, downloaded_size, location):
+        """This method will return True when the disk space available is sufficient
+        for the Download. If not sufficient, it returns False."""
+        download_remaining = file_size - downloaded_size
+        diskspace_available = self.get_availablediskspace(location)
+        # Reserved space can be used to mimic a full disk or prevent filling a disk completely.
+        reserved_space = 0
+        if diskspace_available < (download_remaining + reserved_space):
```

Comment:
This could be simplified. Something like ``return not diskspace_available < (download_remaining + reserved_space)`` should work.